tests/snippets/test_timestamp.py

```diff
@@ -0,0 +1,13 @@
+# flake8: noqa: F841
+
+import pandas as pd
+import datetime as dt
+
+def test_types_init() -> None:
+    ts: pd.Timestamp = pd.Timestamp('2021-03-01T12')
+    ts1: pd.Timestamp = pd.Timestamp(dt.date(2021, 3, 15))
+    ts2: pd.Timestamp = pd.Timestamp(1515590000.1, unit='s')
+    ts3: pd.Timestamp = pd.Timestamp(1515590000.1, unit='s', tz='US/Pacific')
+    ts4: pd.Timestamp = pd.Timestamp(2021, 3, 10, 12)
+    ts5: pd.Timestamp = pd.Timestamp(year=2021, month=3, day=10, hour=12)
+    ts6: pd.Timestamp = pd.Timestamp(year=2021, month=3, day=10, hour=12, tz='US/Pacific')
```

Comment:
A few other possibilities:

```suggestion
    ts6: pd.Timestamp = pd.Timestamp(year=2021, month=3, day=10, hour=12, tz='US/Pacific') 
    ts7: pd.Timestamp = pd.Timestamp(pd.Timestamp('2021-03-01T12'))
    ts8: pd.Timestamp = pd.Timestamp(dt.datetime(2021, 3, 10, 12))
    ts9: pd.Timestamp = pd.Timestamp(1515590000100000000)  # plain integer (nanosecond)
```